### DISCUSSION TITLE
Why does the snap feature not work in pdf

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#76] [General]
Why does the snap feature not work in pdf

**Comment by buggyj** [comment]
Does it not? I can take snaps but the drawing of the cross-hairs to select a part of the screen does not work in firefox. I don't know why.

**Reply by buggyj** [reply]
actual, the reason is that firefox does not allow browser addons to connect to the dom of displayed pdfs.

**Reply by arunnbabu81** [reply]
> Does it not? I can take snaps but the drawing of the cross-hairs to select a part of the screen does not work in firefox. I don't know why.

I also meant that the cross hairs are not appearing to select the screen to be snapped.

**Reply by arunnbabu81** [reply]
> actual, the reason is that firefox does not allow browser addons to connect to the dom of displayed pdfs.

Is there any workarounds for this?

**Comment by arunnbabu81** [comment]
Is there any way to take screenshot from a pdf in firefox and send it to TW

**Comment by buggyj** [comment]
Not at present with tiddlyclip. Firefox plans to enable addons to have access  to images that have been copied to the clipboard but this has not happened yet.
The only current way to snap parts of pdfs is to take a screen shot using the native firefox method, then go to any normal webpage enable content editing and paste the image, and then snap it with tiddlyclip. I do not think this is a good way of working.
Alternatively you could snap the whole screen and then re-snap part of it from within the tiddlywiki.
